src/middlewares/validator.js

```diff
@@ -12,11 +15,31 @@ export default (request, response, next) => {
                 isValid = 'Please provide all fields';
             }
         });
+    } else {
+        isValid = 'No food items';
+    }
         if(isValid === 1) {
             next();
         } else {
             return (response.status(400).json({ status: 'error', message: isValid }));
         }
+    } else if (request.method === 'POST' && (request.originalUrl === '/api/v1/auth/signup' || request.originalUrl === '/api/v1/auth/signup/')) {
+      const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
```

Comment:
Do you think it might be advantageous to move the logic that validates the email to some kind of helper method, so you can re-use for all the places where you have to validate emails? Perhaps you could even do that for other fields.